src/header/mod.rs

```diff
@@ -61,6 +62,14 @@ pub trait HeaderFormat: fmt::Debug + HeaderClone + Any + Typeable + Send + Sync
 
 }
 
+/// A trait for any header value to encode it and send it over the wire
+pub trait EncodeHeader: HeaderFormat + fmt::Display {
+    /// Encode a header to add it into a headers block
+    fn encode_header(&self) -> Vec<Vec<u8>> {
+        vec![format!("{}", self).into_bytes()]
```

Comment:
Since this trait isn't used, it's hard to envision how you expect it to be used. However, I think it'd likely be better to write the bytes directly to a `Write` than to allocate 2 `Vec`s and return them...